datumaro/plugins/voc_format/format.py

```diff
@@ -124,181 +129,278 @@ class VocPath:
     }
 
 
-LabelMapConfig = Dict[str, Tuple[Optional[RgbColor], List[str], List[str]]]
-"""A type representing a label map config"""
-# Not totally type-correct, tuple elements are supposed to support modification.
-# Therefore, the tuple is typically a list
-# TODO: refactor, make type annotations conform with actual usage
+@define
+class VocLabelInfo:
+    color: Optional[RgbColor] = None
+    parts: Set[str] = field(factory=set, converter=set)
+    actions: Set[str] = field(factory=set, converter=set)
 
 
-def make_voc_label_map() -> LabelMapConfig:
-    labels = sorted(VocLabel, key=lambda l: l.value)
-    label_map = OrderedDict((label.name, [VocColormap[label.value], [], []]) for label in labels)
-    label_map[VocLabel.person.name][1] = [p.name for p in VocBodyPart]
-    label_map[VocLabel.person.name][2] = [a.name for a in VocAction]
-    return label_map
+if sys.version_info.minor < 9:
+    _voc_label_map_base = OrderedDict
+else:
+    _voc_label_map_base = OrderedDict[str, VocLabelInfo]
 
 
-def parse_label_map(path: str) -> LabelMapConfig:
+class VocLabelMap(_voc_label_map_base):
+    """
+    Provides VOC-specific info about labels.
+    A mapping of type: str -> VocLabelInfo
     """
-    Parses a label map file in the format:
-    'name : color (r, g, b) : parts (hand, feet, ...) : actions (siting, standing, ...)'
 
-    Parameters:
-        path: File path
+    DEFAULT_BACKGROUND_LABEL = "background"
+    DEFAULT_BACKGROUND_COLOR: RgbColor = (0, 0, 0)
 
-    Returns:
-        A dictionary: label -> (color, parts, actions)
-    """
+    def __init__(self, *args, **kwargs) -> None:
+        d = OrderedDict(*args, **kwargs)  # backward compatibility
```

Comment:
The previous type alias (`LabelMapConfig`) was just an `OrderedDict`, which was typically created from its elements. This class allows this way of creation too, but needs to covert input elements to the correct type.

Reply:
Okay, but there's still no backwards compatibility here, because the calling code had to change anyway (to use the new class).

By the way, does this class really need to support input conversion? It creates inconsistent behavior, where the constructor accepts list values, but `__setitem__` or `update` don't. Seems like the few cases where it's used could be updated to pass `VocLabelInfo` objects instead.

Reply:
I decided to remove backward compatibility mentions and consider this an overload. This class name is new, so its direct clients will need to update their code. For those, who use this class indirectly (as a function output value) the new behavior can be a problem, but I don't think it is important. Old clients, who just passed their labelmaps as dicts into the converter are supported, they don't have to do anything.

Reply:
Alright.

IMO, it would be better to remove the constructor (and thus inherit the base class constructor) and add a separate factory function (`from_legacy_values` or something) that would do the value conversions. That way, dictionary methods would behave consistently, and there would still be an entry point for legacy code. But that's not super important, so I'll leave it up to you.